### DISCUSSION TITLE
Indentation as a proxy for Complexity

### DISCUSSION THREAD

**Discussion by mircealungu** [git-truck/git-truck#853] [General]
One of my friends tried out git-truck with a system in which they have no history. It's a very large system that they have to evaluate and they only have the latest version of the system. I've told him to import it in a local git repo and he could visualize it, but because there's no history information, for such a system git-truck can't show much.

So I've been thinking about metrics that can be computed that are language-independent that could be useful for a large codebase analysis. And I remembered that there was this paper which was showing that indentation can be used as a proxy for complexity [1]. So we could compute an aggregation of indentation-complexity at file level and visualize it maybe. 

Would it make sense to add such a metric? @leonelmerino @joglr @tjomson ? What do you think?

[1] http://svn-plg.uwaterloo.ca/~migod/papers/2008/icpc08-abram.pdf

**Comment by nimrossum** [comment]
Sounds interesting 🤔 This would be really slow to perform, since we then again need to get the contents of all the files in the project (at least the way we did it with Git was slow). We could have an alternative mode that does not use Git at all and just looks at the file system and does analysis such as the one you suggest.

**Comment by mircealungu** [comment]
@Jonas: would that alternative mode not be another tool? 
Maybe it could be a "slow metric" that when you select you get a progress bar that makes you wait while it's being computed?

**Comment by nimrossum** [comment]
Related to #781

**Comment by mircealungu** [comment]
I still think it would be amazing to have this :) 

Could we differentiate between metrics that are already computed and metrics that need to be computed with an UI something like this @nimrossum (lighter gray, dotted underline)?

<img width="460" height="348" alt="image" src="https://github.com/user-attachments/assets/04c0cfd3-9a50-44fd-8b31-e257859abe85" />

Or we could be computing this in the background, and as soon as the metrics become available, we make them available also in the UI. 

If you can implement the UI aspects and scheduling aspects of lazily computed metrics, I could define the metric and the corresponding calculation for complexity.

**Reply by nimrossum** [reply]
Let's do it :)

**Comment by nimrossum** [comment]
That's a really nice idea! 

It talks into the whole progressive enhancement idea as well, for example adding language specific metrics, that could also be lazily loaded!
